tufaceous-lib/src/artifact.rs

```diff
@@ -244,18 +268,39 @@ impl HostPhaseImages {
                 .header()
                 .path()
                 .context("error reading path from archive")?;
-            if path == Path::new(PHASE_1_FILE_NAME) {
+            if path == Path::new(OXIDE_JSON_FILE_NAME) {
+                let json_bytes = read_entry(entry, OXIDE_JSON_FILE_NAME)?;
+                let oxide_json: OxideMetadata =
+                    serde_json::from_slice(&json_bytes).with_context(|| {
+                        format!(
+                        "error deserializing JSON from {OXIDE_JSON_FILE_NAME}"
+                    )
+                    })?;
+
+                let expected = OxideMetadata::os_metadata();
+                if oxide_json != expected {
```

Comment:
It seems like this strict comparison would not allow us to add keys other than the `v` and `t` keys.  Those represent a _minimum_ set of keys, but others are allowed; e.g., we're already adding a dictionary of extra information in a new object-valued `i` property as well.

Reply:
This is fine as it currently stands because we're comparing deserialized formats, and the only keys we're deserializing are `v` and `t`. I'm happy to either make this explicit or add a comment about this, though.

Reply:
Switched to using the metadata impl from helios-omicron-brand which addresses this issue.

Reply:
> This is fine as it currently stands because we're comparing deserialized formats, and the only keys we're deserializing are `v` and `t`. I'm happy to either make this explicit or add a comment about this, though.

Ah ha!  Yes, a comment on that might help.